### DISCUSSION TITLE
do we need to Cache expression compile target?

### DISCUSSION THREAD

**Discussion by ChengHoHang** [schibsted/jslt#265] [Q&A] [answered]
Hi JSLT Team，

for the title,  do we need to Cache **expression compile target**?

actually, my project's jstl script is configed statically not dynamic, and some script is long, if cache this compile target maybe useful?

![image](https://user-images.githubusercontent.com/48877754/186058178-39ea4ced-1794-420c-ae9f-aced485f7b77.png)

some project have Cache compilation, such as mybatis will save sql tree nodes in memory, is jstl also support?

**Comment by larsga** [comment, marked as answer]
I'm not sure I understand the question, but if you want to avoid compiling the `template` over and over again, you can simply store the `Expression` object and reuse it. Then you parse once, and use the JSLT expression many times.

**Reply by ChengHoHang** [reply]
yes, what you said is somthing that i'm going to do,  i just want to confirm whether `compile()` would cache my `template` :)

**Reply by larsga** [reply]
Well, it's not going to cache the `template` string. But the same `Expression` object can be used many times with different input.
